src/main/java/com/onarandombox/MultiverseCore/listeners/MVPlayerListener.java

```diff
@@ -144,87 +167,54 @@ public void playerChangedWorld(PlayerChangedWorldEvent event) {
      */
     @EventHandler(priority = EventPriority.HIGHEST)
     public void playerTeleport(PlayerTeleportEvent event) {
-        Logging.finer("Got teleport event for player '"
-                + event.getPlayer().getName() + "' with cause '" + event.getCause() + "'");
-        if (event.isCancelled()) {
+        if (event.isCancelled() || event.getTo() == null) {
             return;
         }
+
+        Logging.fine("Got teleport event for player '"
+                + event.getPlayer().getName() + "' with cause '" + event.getCause() + "'");
+
         Player teleportee = event.getPlayer();
         CommandSender teleporter = null;
+
+        //TODO - Redo teleporter queue handling
         String teleporterName = MultiverseCore.getPlayerTeleporter(teleportee.getName());
-        if (teleporterName != null) {
-            if (teleporterName.equals("CONSOLE")) {
-                Logging.finer("We know the teleporter is the console! Magical!");
-                teleporter = this.plugin.getServer().getConsoleSender();
-            } else {
-                teleporter = this.plugin.getServer().getPlayerExact(teleporterName);
-            }
-        }
-        Logging.finer("Inferred sender '" + teleporter + "' from name '"
-                + teleporterName + "', fetched from name '" + teleportee.getName() + "'");
-        MVWorld fromWorld = this.worldManager.getMVWorld(event.getFrom().getWorld().getName());
-        MVWorld toWorld = this.worldManager.getMVWorld(event.getTo().getWorld().getName());
-        if (toWorld == null) {
-            Logging.fine("Player '" + teleportee.getName() + "' is teleporting to world '"
-                    + event.getTo().getWorld().getName() + "' which is not managed by Multiverse-Core.  No further "
-                    + "actions will be taken by Multiverse-Core.");
+        if (teleporterName == null && !this.plugin.getMVConfig().getTeleportIntercept()) {
+            Logging.finer("Teleporting without intercepting.");
             return;
         }
-        if (event.getFrom().getWorld().equals(event.getTo().getWorld())) {
-            // The player is Teleporting to the same world.
-            Logging.finer("Player '" + teleportee.getName() + "' is teleporting to the same world.");
-            this.stateSuccess(teleportee.getName(), toWorld.getAlias());
-            return;
+
+        if (Objects.equals(teleporterName, "CONSOLE")) {
+            Logging.finer("We know the teleporter is the console! Magical!");
+            teleporter = this.plugin.getServer().getConsoleSender();
+        } else {
+            teleporter = this.plugin.getServer().getPlayerExact(teleporterName);
         }
-        // TODO: Refactor these lines.
-        // Charge the teleporter
-        event.setCancelled(!pt.playerHasMoneyToEnter(fromWorld, toWorld, teleporter, teleportee, true));
```

Comment:
Was this charging the player and also the cause of the double charge bug?

Reply:
yea it was, bcu for some reason the method `playerHasMoneyToEnter` also contain the actual charging of the player, which confused me.